src/main/scala/com/cloudant/clouseau/IndexService.scala

```diff
@@ -734,9 +734,10 @@ class IndexService(ctx: ServiceContext[IndexServiceArgs]) extends Service(ctx) w
     if (HPs != null) {
       val highlights = (HPs.highlightFields zip HPs.analyzers).map {
         case (field, analyzer) =>
-          val frags = HPs.highlighter.getBestFragments(analyzer, field,
-            doc.get(field), HPs.highlightNumber).toList
-          (field, frags)
+          (field, doc.getValues(field).flatMap { v =>
```

Comment:
I like better the original approach of using the additional `frags` variable -- in my humble opinion this makes the code easier to read.